src/Integration/Spiral/Blade/Bootloader/BladeBootloader.php

```diff
@@ -91,25 +102,28 @@ public function boot(Container $container): void
             return $engineResolver;
         });
 
-        $container->bindSingleton('blade.dispatcher', function (Container $container) {
-            return new Dispatcher();
-        });
-
-        $container->bindSingleton('blade', function (Container $container) {
-            return new Factory(
-                $container->get('blade.engine_resolver'),
-                $container->get('blade.file_view_finder'),
-                $container->get('blade.dispatcher'),
-            );
-        });
-
-        $container->bind(Factory::class, 'blade');
-
-        $container->bindSingleton('schranz_templating.renderer.blade', function (Container $container) {
-            return new BladeRenderer($container->get('blade'));
-        });
+        $binder->bindSingleton(DispatcherContract::class, Dispatcher::class);
+
+        $binder->bindSingleton(
+            FactoryContract::class,
+            static function (
+                EngineResolver $resolver,
+                ViewFinderInterface $finder,
+                DispatcherContract $dispatcher
+            ): FactoryContract {
+                return new Factory(
+                    $resolver,
+                    $finder,
+                    $dispatcher,
+                );
+            }
+        );
 
-        $container->bind(TemplateRendererInterface::class, 'schranz_templating.renderer.blade');
-        $container->bind(BladeRenderer::class, 'schranz_templating.renderer.blade');
+        $binder->bindSingleton(
+            TemplateRendererInterface::class,
+            static function (FactoryContract $factory): TemplateRendererInterface {
+                return new BladeRenderer($factory);
+            }
+        );
```

Comment:
The integration will always provide the specific Renderer as a service. But also bind the Interface to that service.

```suggestion
        $binder->bindSingleton(
            BladeRenderer::class,
            static function (FactoryContract $factory): TemplateRendererInterface {
                return new BladeRenderer($factory);
            }
        );

        $binder->bind(TemplateRendererInterface::class, BladeRenderer::class);
```

This way we make sure somebody can inject and use the specific renderer, but a library/module can access the project defined renderer via TemplateRendererInterface.